---
title: "01_Map_Reads"
author: "Julin Maloof"
date: "2023-12-07"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# 01 Map Reads

Script to map the RNAseq reads to a genomic reference.  I will use John's S. div Hifi ASM.  

## Assembly info

Assembly is on Box at https://ucdavis.app.box.com/folder/190583930929

Assembly is on LSSC0 at `/share/malooflab/ref_genomes/S_diversifolious/Sdiv.2023HiFiasm`

## Reads info

Apex libraries were sequenced in three pools.  There was an error made in sequencing Apex1--it was combined with another library that had incompatible indexes, so many (but not all) samples were clobbered.  It was resequenced later.

### Location of Apex2 and Apex3 round 1 reads:

  /share/malooflab/Fastqs/Dimensions/hydrothermal_2022-09-20
  # Note there are seed reads in there as well, so take care when specifying files
  
These reads have file names like `Apex3_yo1-24_i0_S227_L002_R1_001.fastq.gz`

Retrieve read 1 files with:
```{bash}
find /share/malooflab/Fastqs/Dimensions/hydrothermal_2022-09-20 | grep -E "Apex.*_R1_.*fastq\.gz$" 
```


### Location of Apex1 round 1 reads:

  /share/malooflab/Fastqs/Dimensions/hydrothermal_2022-09-20-apex1-redo/good
  
These reads have file names like `r1.bhr-02_i0.fq.gz`

Rename these to move read name to end, like other files:

```{bash}
cd /share/malooflab/Fastqs/Dimensions/hydrothermal_2022-09-20-apex1-redo/good

for f in $(ls)
  do
    newname=$(echo $f | sed -E "s/r(1|2)\.(.*)\.fq.gz/\\2_R\\1_.fastq.gz/")
    echo "old name: "$f
    echo "new name: "$newname
    mv $f $newname
  done
```

Retrieve read 1 files with

```{bash}
find /share/malooflab/Fastqs/Dimensions/hydrothermal_2022-09-20-apex1-redo/good | grep -E "_R1_.fastq.gz" 
```

### Location of round 2 reads:

  /share/malooflab/Fastqs/Dimensions/hydrothermal_apex_2023-07-24/FASTQ_0292
  /share/malooflab/Fastqs/Dimensions/hydrothermal_apex_2023-07-24/FASTQ_0293
  
These reads have file names like `72_JM72_9999RND_tm2-01-v1_RNAseq_0_77-C_S72_L001_R1_001.fastq.gz`

Retrieve read 1 files with:

```{bash}
find   /share/malooflab/Fastqs/Dimensions/hydrothermal_apex_2023-07-24/ | grep -E "_[a-z]{2}[a-z0-9]-[0-9]{2}.*_R1_.*fastq\.gz$" 
```

## Create File List

List of Read 1 files for mapping.
```{bash}
cd /share/malooflab/Julin/git/S_tort-vernalization/intermediate

find /share/malooflab/Fastqs/Dimensions/hydrothermal_2022-09-20-apex1-redo/good | grep -E "_R1_.fastq.gz" > R1_fastqs.txt


find /share/malooflab/Fastqs/Dimensions/hydrothermal_2022-09-20 -maxdepth 1 | grep -E "Apex.*_R1_.*fastq\.gz$"  >> R1_fastqs.txt

find /share/malooflab/Fastqs/Dimensions/hydrothermal_apex_2023-07-24/ | grep -E "_[a-z]{2}[a-z0-9]-[0-9]{2}.*_R1_.*fastq\.gz$" >> R1_fastqs.txt


```


## List of failed:

first get list of worked:

```{bash}
cd /share/malooflab/Julin/git/S_tort-vernalization/intermediate/bams
ls -1 *final* | sed s/_Log.final.out// > ../worked.txt
cd ../
grep -f worked.txt -v R1_fastqs.txt > R1_fastqs_redo.txt
 
```


See SLURM script

## List of files for htseq-count

```{bash}
cd /share/malooflab/Julin/git/S_tort-vernalization/intermediate/bams
ls -1 *Aligned.sortedByCoord.out.bam > ../bams_to_count.txt

```


