---
title: "temporary Vernalization 04 file"
author: "XinYu Zhang"
date: "2023-11-08"
output: html_document
---



Exercise 1. Loading the dataset (raw_data)
```{r}
raw_data<-read.csv("~/Desktop/raw_counts.csv.gz")
```


Exercise 3.Fix the column names by removing "." because “.” is a wildcard character.
```{r}
library(stringr)
colnames(raw_data) <- str_remove(colnames(raw_data), fixed("."))
colnames(raw_data)
```


Exercise 4.Make a histogram of counts for each of the samples
Conclusion: the data is not normally distributed
```{r}
library(tidyr)
library(ggplot2)
library(dplyr)
#change data into long format
raw_data_long_format <- raw_data %>% 
  pivot_longer(c(-geneID), names_to="sample", values_to="count")

#Make a histogram
ggplot(raw_data_long_format, aes(x = count)) +
  geom_histogram(bins = 10, fill = "blue", color = "black") +
  xlim(c(0, 1000)) + 
  theme_minimal() +
  labs(title = "Histogram of Counts within Specified Range", x = "Count", y = "Frequency")
#Conclusion: Non-normal distribution

```

```{r}
raw_data <- raw_data[rowSums(raw_data[,-1] > 10) >= 3,]#For our subsequent analyses we want to reduce the data set to only those genes with some expression. In this case we will retain genes with > 10 reads in >= 3 samples
```

Exercise 5.check for correlation and visualize it
```{r}
library(tidyr)
library(dplyr)
library(gplots)
library(viridis)
library(stringr)

raw_data_limited <- raw_data[1:1000, -which(colnames(raw_data) == "geneID")]
# Adjust margins: c(bottom, left, top, right)
par(mar=c(4, 4, 2, 2))
#pairs(raw_data_limited) Large gra
raw_data_cor_table <- cor(raw_data_limited, use = "complete.obs")
print(raw_data_cor_table)#correlation table



raw_data_cor_table %>% gplots::heatmap.2(dendrogram="row", trace = "none", col=viridis::viridis(25, begin=.25), margins=c(7,8))
```


Exercise 6.Data Normalization
```{r}
sample.description_raw_data <-tibble(sample=colnames(raw_data)[-1])

data <- sample.description_raw_data%>% mutate(
    Pop = str_extract(sample,"bhr|ihl|kc|lv|sq|tm|wl|yo"), Vern = str_extract(sample, "v0|v1|v2|i0|i1|i2"),
    group = str_c(Vern, Pop, sep = "_"))
print(head(data))

#convert Elevation and Vernalization into factors for testing No vern,short vern, and long vern

sample.description_data <- data%>%
  mutate(Pop=factor(Pop), 
         Vern=factor(Vern,levels = c("v0","v1", "v2","i0","i1","i2"))) # setting the levels in this way makes "v0" the reference  

sample.description_data
```


Exercise 6. Normalization factor and plot
```{r}
#calculate the effective library size and normalization factors using the TMM method
library(edgeR)

counts.matrix_raw_data <- raw_data %>% select(-geneID) %>% as.matrix()
rownames(counts.matrix_raw_data) <- raw_data$geneID

dge.data_Normalization <- DGEList(counts=counts.matrix_raw_data, 
                    group=sample.description_raw_data$group)
dim(dge.data_Normalization) 
dge.data_Normalization <- calcNormFactors(dge.data_Normalization, method = "TMM")
dge.data_Normalization$sample # look at the normalization factors

plotMDS(dge.data_Normalization, method = "bcv") 
```

```{r}
## make sure sample order is the same in the dge object and in the sample sheet
all(sample.description_data$sample==rownames(dge.data_Normalization$samples) )
```

subset bhr
```{r}
dge.data_Normalization.bhr <- dge.data_Normalization[,sample.description_data$Pop=="bhr"]
dge.data_Normalization.bhr$samples
```


Exercise 8
```{r}
data_Normalization<-cpm(dge.data_Normalization)
data_Normalization_log <- log2(data_Normalization[,-1]+1)

design <- model.matrix(~Pop+Vern,data = sample.description_data)
rownames(design) <- sample.description_data$sample
design


dge.data_Normalization_dispersions <- estimateGLMCommonDisp(dge.data_Normalization,design,verbose = TRUE)
dge.data_Normalization_dispersions <- estimateGLMTrendedDisp(dge.data_Normalization_dispersions,design)
dge.data_Normalization_dispersions <-estimateGLMTagwiseDisp(dge.data_Normalization_dispersions,design)
plotBCV(dge.data_Normalization_dispersions) 
```

```{r}
fit <- glmFit (dge.data_Normalization_dispersions,design)
gt.lrt <- glmLRT(fit,coef="Vernv1")
topTags(gt.lrt)
summary(decideTestsDGE(gt.lrt,p.value=0.01))

Pop_Vern_gene <- topTags(gt.lrt,n = Inf,p.value = 0.01)$table
Pop_Vern_gene

```


Transcriptional
```{r}
library(tibble)
plotDE <- function(genes, dge.data_Normalization_dispersions, sample.description_raw_data) {
  require(ggplot2)
  tmp.data <- t(log2(cpm(dge.data_Normalization_dispersions[genes,])+1))
  tmp.data <- tmp.data %>%
    as.data.frame() %>%
    rownames_to_column("sample") %>%
    left_join(sample.description_raw_data,by="sample")
  tmp.data <- tmp.data %>%
    pivot_longer(cols=starts_with("Sdiv"), values_to = "log2_cpm", names_to = "gene")
  pl <- ggplot(tmp.data,aes(x=Vern,y=log2_cpm, fill=Pop))
  pl <- pl + facet_wrap( ~ gene)
  pl <- pl + ylab("log2(cpm)") + xlab("genotype")
  pl <- pl + geom_boxplot()
  pl + theme(axis.text.x  = element_text(angle=45, vjust=1,hjust=1))
}

plotDE("Sdiv_ptg000001l_0001-R",dge.data_Normalization_dispersions,sample.description_data)
plotDE(rownames(Pop_Vern_gene)[1:3],dge.data_Normalization_dispersions,sample.description_data)


```



subset bhr single comparsion v0 vs v1
```{r}
dge.data_Normalization.bhr <- dge.data_Normalization [,sample.description_data $Pop=="bhr"]
dge.data_Normalization.bhr$samples

sample.description_data.bhr <- sample.description_data %>%
  filter(Pop == "bhr")


design.bhr <- model.matrix(~Vern,data = sample.description_data.bhr)
rownames(design.bhr) <- sample.description_data.bhr$samples
design.bhr


dge.data_Normalization_dispersions.bhr <- estimateGLMCommonDisp(dge.data_Normalization.bhr,design.bhr,verbose = TRUE)

fit.bhr <- glmFit(dge.data_Normalization_dispersions.bhr,design.bhr)
gt.lrt.bhr <- glmLRT(fit.bhr,coef="Vernv1")
result_v0v1_bhr<-summary(decideTestsDGE(gt.lrt.bhr,p=0.01))
print(result_v0v1_bhr)
```


subset yo single comparsion 
```{r}
dge.data_Normalization.yo <- dge.data_Normalization [,sample.description_data $Pop=="yo"]
dge.data_Normalization.yo$samples

sample.description_data.yo <- sample.description_data %>%
  filter(Pop == "yo")
sample.description_data.yo

design.yo <- model.matrix(~Vern,data = sample.description_data.yo)
rownames(design.yo) <- sample.description_data.yo$samples
design.yo


dge.data_Normalization_dispersions.yo <- estimateGLMCommonDisp(dge.data_Normalization.yo,design.yo,verbose = TRUE)

fit.yo <- glmFit(dge.data_Normalization_dispersions.yo,design.yo)
gt.lrt.yo <- glmLRT(fit.yo,coef="Vernv1")
result_v0v1.yo<-summary(decideTestsDGE(gt.lrt.yo,p=0.01))

print(result_v0v1.yo)
```

Loop for each population (vo vs v1)


| Population | Down | NotSig | Up   |
|------------|------|--------|------|
| bhr        | 298  | 29088  | 597  |
| ihl        | 203  | 28977  | 803  |
| kc         | 282  | 28978  | 723  |
| lv         | 203  | 28458  | 1322 |
| sq         | 637  | 29302  | 44   |
| tm         | 1521 | 27924  | 538  |
| wl         | 112  | 29533  | 338  |
| yo         | 270  | 29373  | 340  |


```{r}
pop_groups <- unique(sample.description_data$Pop)
results_list <- list()

for (pop in pop_groups) {

  #Extract the Normalization data for each population
  dge.data_Normalization_pop <- dge.data_Normalization[, sample.description_data$Pop == pop]
  #Extract sample description data for each population
  sample.description_data_pop <- sample.description_data %>% filter(Pop == pop)
  #redesign the matrix with correct # column for each population
  design_pop <- model.matrix(~Vern, data = sample.description_data_pop)
  rownames(design_pop) <- sample.description_data_pop$samples
  #using matrix to find out the Normalization dispersions for each population
  dge.data_Normalization_dispersions_pop <- estimateGLMCommonDisp(dge.data_Normalization_pop, design_pop, verbose = TRUE)
  
  #Fit the model for each population dispersion data
  fit_pop <- glmFit(dge.data_Normalization_dispersions_pop, design_pop)
  # Conduct likelihood ratio test for each population
  gt.lrt_pop <- glmLRT(fit_pop, coef = "Vernv1")
  
  # Summarize the results for each population (up / down regulation)
  result_v0v1_pop <- summary(decideTestsDGE(gt.lrt_pop, p = 0.01))
  description <- "v0v1 comparison for population: " 
  results_list[[pop]] <- list(
  Description = paste(description, pop, sep=""),
  Results = result_v0v1_pop)
  }

# Print or return results
print(results_list)
```




Loop for each population (vo vs v2)


| Population | Down | NotSig | Up   |
|------------|------|--------|------|
| bhr        | 530  | 28489  | 964  |
| ihl        | 368  | 28630  | 985  |
| kc         | 105  | 29197  | 681  |
| lv         | 1081 | 27785  | 1117 |
| sq         | 707  | 29161  | 115  |
| tm         | 35   | 29708  | 240  |
| wl         | 223  | 29277  | 483  |
| yo         | 126  | 29297  | 560  |

```{r}
pop_groups <- unique(sample.description_data$Pop)
results_list <- list()

for (pop in pop_groups) {

  #Extract the Normalization data for each population
  dge.data_Normalization_pop <- dge.data_Normalization[, sample.description_data$Pop == pop]
  #Extract sample description data for each population
  sample.description_data_pop <- sample.description_data %>% filter(Pop == pop)
  #redesign the matrix with correct # column for each population
  design_pop <- model.matrix(~Vern, data = sample.description_data_pop)
  rownames(design_pop) <- sample.description_data_pop$samples
  #using matrix to find out the Normalization dispersions for each population
  dge.data_Normalization_dispersions_pop <- estimateGLMCommonDisp(dge.data_Normalization_pop, design_pop, verbose = TRUE)
  
  #Fit the model for each population dispersion data
  fit_pop <- glmFit(dge.data_Normalization_dispersions_pop, design_pop)
  # Conduct likelihood ratio test for each population
  gt.lrt_pop <- glmLRT(fit_pop, coef = "Vernv2")
  
  # Summarize the results for each population (up / down regulation)
  result_v0v2_pop <- summary(decideTestsDGE(gt.lrt_pop, p = 0.01))
  description <- "v0v2 comparison for population: " 
  results_list[[pop]] <- list(
  Description = paste(description, pop, sep=""),
  Results = result_v0v2_pop)
  }

# Print or return results
print(results_list)

```

Loop for each population (v1 vs v2)

| Population | Down | NotSig | Up   |
|------------|------|--------|------|
| bhr        | 468  | 29029  | 486  |
| ihl        | 374  | 29304  | 305  |
| kc         | 19   | 29778  | 186  |
| lv         | 1220 | 28555  | 208  |
| sq         | 195  | 29605  | 183  |
| tm         | 0    | 29940  | 43   |
| wl         | 108  | 29689  | 186  |
| yo         | 87   | 29465  | 431  |


1. create a new reference for v1
```{r}
sample.description_data_v1_reference <- data%>%
  mutate(Pop=factor(Pop), 
         Vern=factor(Vern,levels = c("v1","v0", "v2","i0","i1","i2"))) # setting the levels in this way makes "v1" the reference  

sample.description_data_v1_reference
```

2. loop for comparison
```{r}
pop_groups <- unique(sample.description_data_v1_reference$Pop)
results_list <- list()

for (pop in pop_groups) {

  #Extract the Normalization data for each population
  dge.data_Normalization_pop <- dge.data_Normalization[, sample.description_data_v1_reference$Pop == pop]
  #Extract sample description data for each population
  sample.description_data_pop <- sample.description_data_v1_reference %>% filter(Pop == pop)
  #redesign the matrix with correct # column for each population
  design_pop <- model.matrix(~Vern, data = sample.description_data_pop)
  rownames(design_pop) <- sample.description_data_pop$samples
  #using matrix to find out the Normalization dispersions for each population
  dge.data_Normalization_dispersions_pop <- estimateGLMCommonDisp(dge.data_Normalization_pop, design_pop, verbose = TRUE)
  
  #Fit the model for each population dispersion data
  fit_pop <- glmFit(dge.data_Normalization_dispersions_pop, design_pop)
  # Conduct likelihood ratio test for each population
  gt.lrt_pop <- glmLRT(fit_pop, coef = "Vernv2")
  # Summarize the results for each population (up / down regulation)
  result_v1v2_pop <- summary(decideTestsDGE(gt.lrt_pop, p = 0.01))
  description <- "v1v2 comparison for population: " 
  results_list[[pop]] <- list(
    Description = paste(description, pop, sep=""),
    Results = result_v1v2_pop)
   }

# Print or return results
print(results_list)
```


Full Table

```{r}
# Load the necessary library
library(dplyr)

# Define the data frames for each comparison
vo_vs_v1_table <- data.frame(
  Population = c("bhr", "ihl", "kc", "lv", "sq", "tm", "wl", "yo"),
  Down = c(298, 203, 282, 203, 637, 1521, 112, 270),
  NotSig = c(29088, 28977, 28978, 28458, 29302, 27924, 29533, 29373),
  Up = c(597, 803, 723, 1322, 44, 538, 338, 340),
  Comparison = 'vo_vs_v1',
  Elevation = c("low","low","low","high","high","low","high","high")
)

vo_vs_v2_table<- data.frame(
  Population = c("bhr", "ihl", "kc", "lv", "sq", "tm", "wl", "yo"),
  Down = c(530, 368, 105, 1081, 707, 35, 223, 126),
  NotSig = c(28489, 28630, 29197, 27785, 29161, 29708, 29277, 29297),
  Up = c(964, 985, 681, 1117, 115, 240, 483, 560),
  Comparison = 'vo_vs_v2',
  Elevation = c("low","low","low","high","high","low","high","high")
)

v1_vs_v2_table <- data.frame(
  Population = c("bhr", "ihl", "kc", "lv", "sq", "tm", "wl", "yo"),
  Down = c(468, 374, 19, 1220, 195, 0, 108, 87),
  NotSig = c(29029, 29304, 29778, 28555, 29605, 29940, 29689, 29465),
  Up = c(486, 305, 186, 208, 183, 43, 186, 431),
  Comparison = 'v1_vs_v2',
  Elevation = c("low","low","low","high","high","low","high","high")
)


combined_full_data_table <- bind_rows(vo_vs_v1_table, vo_vs_v2_table, v1_vs_v2_table)

write.csv(combined_full_data_table, "combined_full_data_table_v0v1v2.csv", row.names = FALSE)


print(combined_full_data_table)

```

Histogram of the Full
```{r}
# Load necessary libraries
library(ggplot2)
library(dplyr)


# Read the data
combined.full.histogram.data <- read.csv("combined_full_data_table_v0v1v2.csv")

# Melt the data to long format for easier plotting
combined.full.histogram.plot <- combined.full.histogram.data %>%
  gather(key = "Regulation", value = "Count", -Population, -Comparison,-Elevation)

# Create a new variable for easier labeling in the plot
combined.full.histogram.plot$Population_Comparison <- paste(combined.full.histogram.plot$Population, combined.full.histogram.plot$Comparison, combined.full.histogram.plot$Elevation,sep = "/")

combined.full.histogram.plot$Population_Comparison <- gsub("v0_vs_v1", "v0v1", combined.full.histogram.plot$Population_Comparison)
combined.full.histogram.plot$Population_Comparison <- gsub("v0_vs_v2", "v0v2", combined.full.histogram.plot$Population_Comparison)
combined.full.histogram.plot$Population_Comparison <- gsub("v1_vs_v2", "v1v2", combined.full.histogram.plot$Population_Comparison)

ggplot(combined.full.histogram.plot, aes(x = Population_Comparison, y = Count, fill = Regulation)) +
  geom_bar(stat = "identity", position = position_dodge(width = 0.7)) +  
  geom_text(aes(label = Count), vjust = -0.3, position = position_dodge(width = 0.9)) + 
  labs(title = "Gene Regulation Across Populations and Comparisons",
       x = "Population (Comparison)",
       y = "Count of Genes") +
  theme_minimal() +
  scale_fill_brewer(palette = "Dark2") +  # Set a color palette
  theme(axis.text.x = element_text(angle = 45, hjust = 1))  # Rotate x-axis labels for better visibility

```


```{r}

Group_1_tm_lv_data <- filter(combined.full.histogram.data, Population %in% c("tm", "lv"))
Group_2_ihl_wl_data <- filter(combined.full.histogram.data, Population %in% c("ihl", "wl"))
Group_3_bhr_yo_data <- filter(combined.full.histogram.data, Population %in% c("bhr", "yo"))
Group_4_kc_sq_data <- filter(combined.full.histogram.data, Population %in% c("kc", "sq"))

```

```{r}
# Load necessary libraries
library(ggplot2)
library(dplyr)
library(tidyr)  # for gather function

# Assuming Group_1_tm_lv_data has already been created by filtering combined.full.histogram.data
# Melt the data to long format for easier plotting
group_1_plot <- Group_1_tm_lv_data %>%
  gather(key = "Regulation", value = "Count", -Population, -Comparison, -Elevation)

# Create a new variable for easier labeling in the plot
group_1_plot$Population_Comparison <- paste(group_1_plot$Population, group_1_plot$Comparison, group_1_plot$Elevation, sep = "/")

# Simplify labels if necessary (depending on your specific labeling needs)
group_1_plot$Population_Comparison <- gsub("v0_vs_v1", "v0v1", group_1_plot$Population_Comparison)
group_1_plot$Population_Comparison <- gsub("v0_vs_v2", "v0v2", group_1_plot$Population_Comparison)
group_1_plot$Population_Comparison <- gsub("v1_vs_v2", "v1v2", group_1_plot$Population_Comparison)

# Plotting
ggplot(group_1_plot, aes(x = Population_Comparison, y = Count, fill = Regulation)) +
  geom_bar(stat = "identity", position = position_dodge(width = 0.7)) +  
  geom_text(aes(label = Count), vjust = -0.3, position = position_dodge(width = 0.9)) + 
  labs(title = "Gene Regulation in Group 1 (Populations tm and lv)",
       x = "Population (Comparison/Elevation)",
       y = "Count of Genes") +
  theme_minimal() +
  scale_fill_brewer(palette = "Dark2") +  # Set a color palette
  theme(axis.text.x = element_text(angle = 45, hjust = 1))  # Rotate x-axis labels for better visibility

```

```{r}
# Load necessary libraries
library(ggplot2)
library(dplyr)
library(tidyr)  # for gather function

# Assuming Group_2_ihl_wl_data has already been created by filtering combined.full.histogram.data
# Melt the data to long format for easier plotting
group_2_plot <- Group_2_ihl_wl_data %>%
  gather(key = "Regulation", value = "Count", -Population, -Comparison, -Elevation)

# Create a new variable for easier labeling in the plot
group_2_plot$Population_Comparison <- paste(group_2_plot$Population, group_2_plot$Comparison, group_2_plot$Elevation, sep = "/")

# Simplify labels if necessary (depending on your specific labeling needs)
group_2_plot$Population_Comparison <- gsub("v0_vs_v1", "v0v1", group_2_plot$Population_Comparison)
group_2_plot$Population_Comparison <- gsub("v0_vs_v2", "v0v2", group_2_plot$Population_Comparison)
group_2_plot$Population_Comparison <- gsub("v1_vs_v2", "v1v2", group_2_plot$Population_Comparison)

# Plotting
ggplot(group_2_plot, aes(x = Population_Comparison, y = Count, fill = Regulation)) +
  geom_bar(stat = "identity", position = position_dodge(width = 0.7)) +  
  geom_text(aes(label = Count), vjust = -0.3, position = position_dodge(width = 0.9)) + 
  labs(title = "Gene Regulation in Group 2 (Populations ihl and wl)",
       x = "Population (Comparison/Elevation)",
       y = "Count of Genes") +
  theme_minimal() +
  scale_fill_brewer(palette = "Dark2") +  # Set a color palette
  theme(axis.text.x = element_text(angle = 45, hjust = 1))  # Rotate x-axis labels for better visibility

```


```{r}
# Assuming Group_3_bhr_yo_data has already been created by filtering combined.full.histogram.data
# Melt the data to long format for easier plotting
group_3_plot <- Group_3_bhr_yo_data %>%
  gather(key = "Regulation", value = "Count", -Population, -Comparison, -Elevation)

# Create a new variable for easier labeling in the plot
group_3_plot$Population_Comparison <- paste(group_3_plot$Population, group_3_plot$Comparison, group_3_plot$Elevation, sep = "/")

# Simplify labels if necessary
group_3_plot$Population_Comparison <- gsub("v0_vs_v1", "v0v1", group_3_plot$Population_Comparison)
group_3_plot$Population_Comparison <- gsub("v0_vs_v2", "v0v2", group_3_plot$Population_Comparison)
group_3_plot$Population_Comparison <- gsub("v1_vs_v2", "v1v2", group_3_plot$Population_Comparison)

# Plotting
ggplot(group_3_plot, aes(x = Population_Comparison, y = Count, fill = Regulation)) +
  geom_bar(stat = "identity", position = position_dodge(width = 0.7)) +  
  geom_text(aes(label = Count), vjust = -0.3, position = position_dodge(width = 0.9)) + 
  labs(title = "Gene Regulation in Group 3 (Populations bhr and yo)",
       x = "Population (Comparison/Elevation)",
       y = "Count of Genes") +
  theme_minimal() +
  scale_fill_brewer(palette = "Dark2") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

```



```{r}
# Assuming Group_4_kc_sq_data has already been created by filtering combined.full.histogram.data
# Melt the data to long format for easier plotting
group_4_plot <- Group_4_kc_sq_data %>%
  gather(key = "Regulation", value = "Count", -Population, -Comparison, -Elevation)

# Create a new variable for easier labeling in the plot
group_4_plot$Population_Comparison <- paste(group_4_plot$Population, group_4_plot$Comparison, group_4_plot$Elevation, sep = "/")

# Simplify labels if necessary
group_4_plot$Population_Comparison <- gsub("v0_vs_v1", "v0v1", group_4_plot$Population_Comparison)
group_4_plot$Population_Comparison <- gsub("v0_vs_v2", "v0v2", group_4_plot$Population_Comparison)
group_4_plot$Population_Comparison <- gsub("v1_vs_v2", "v1v2", group_4_plot$Population_Comparison)

# Plotting
ggplot(group_4_plot, aes(x = Population_Comparison, y = Count, fill = Regulation)) +
  geom_bar(stat = "identity", position = position_dodge(width = 0.7)) +  
  geom_text(aes(label = Count), vjust = -0.3, position = position_dodge(width = 0.9)) + 
  labs(title = "Gene Regulation in Group 4 (Populations kc and sq)",
       x = "Population (Comparison/Elevation)",
       y = "Count of Genes") +
  theme_minimal() +
  scale_fill_brewer(palette = "Dark2") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

```

